### PR TITLE
Allow discount and product tax code to send along in Taxjar Api Request

### PR DESCRIPTION
**Pull request by pikender**
#18

### CODE REVIEW

File: app/models/spree/taxjar.rb [hunk 1]
```diff
@@ -72,11 +72,15 @@ def tax_params
 
       def taxable_line_items_params
         @order.line_items.map do |item|
-          {
+          h = {
```
Comment (nishant-cyro) [inline app/models/spree/taxjar.rb]: @pikender - I think we should use more meaningful variable names here.

File: lib/spree_taxjar/categories.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          p.fetch(:product_tax_code)
+        end
+
+        def description(p)
+          p.fetch(:description)
+        end
+
+        def default_categories
+          [
+            {:name=>"Digital Goods", :product_tax_code=>"31000", :description=>"Digital products transferred electronically, meaning obtained by the purchaser by means other than tangible storage media."},
```
Comment (nishant-cyro) [inline lib/spree_taxjar/categories.rb]: @pikender - Let's use new syntax of hash here.

Comment (pikender) [inline lib/spree_taxjar/categories.rb]: @nishant-cyro can we skip the new hash syntax as this is generated via a script ( which fetches available product categories using taxjar api ) whose output is copied here and ruby prints the hash in rocket_syntax (=>) whether defined like json or rocket_syntax 😞

File: app/models/spree/taxjar.rb (hunk 1 above)
Comment (pikender) [inline app/models/spree/taxjar.rb]: @nishant-cyro how about line_item_attrs or line_item_attributes ??

Comment (nishant-cyro) [inline app/models/spree/taxjar.rb]: line_item_attributes seems fine to me.

File: lib/spree_taxjar/categories.rb (hunk 2 above)
Comment (nishant-cyro) [inline lib/spree_taxjar/categories.rb]: ok.

File: app/models/spree/taxjar.rb (hunk 1 above)
Comment (pikender) [inline app/models/spree/taxjar.rb]: 👍

File: lib/spree_taxjar/categories.rb (hunk 2 above)
Comment (pikender) [inline lib/spree_taxjar/categories.rb]: Thanks 😊

**Comment by pikender** [comment]
The changes are consolidated and shared in #25

**Pull request closed without merging**